File: apps/admin/src/api/configitem.ts
import { apiFetch } from './http'
import { buildListQuery } from './query'
import type {
  ConfigItemCreateRep,
  ConfigItemCreateReq,
  ConfigItemListRep,
  ConfigItemListReq,
  ConfigItemMain,
  ConfigItemUpdateReq,
} from './types'

/**
 * REST client for the kusec `ConfigItem` service (see
 * api/proto/kusec_v1/configitem.proto). Routes are singular per the gateway
 * contract: `/config-item`, `/config-item/{id}`. A ConfigItem is scoped to a
 * ConfigMap via `configmap_id`.
 */

export function listConfigItems(
  req: ConfigItemListReq = {},
): Promise<ConfigItemListRep> {
  const query = buildListQuery(req.list_params, {
    configmap_id: req.configmap_id,
    configmap_ids: req.configmap_ids,
    active: req.active,
    search: req.search,
  })
  return apiFetch<ConfigItemListRep>(`/config-item${query}`)
}

export function getConfigItem(id: string): Promise<ConfigItemMain> {
  return apiFetch<ConfigItemMain>(`/config-item/${encodeURIComponent(id)}`)
}

export function createConfigItem(
  req: ConfigItemCreateReq,
): Promise<ConfigItemCreateRep> {
  return apiFetch<ConfigItemCreateRep>('/config-item', {
    method: 'POST',
    body: JSON.stringify(req),
  })
}

export function updateConfigItem(
  id: string,
  req: ConfigItemUpdateReq,
): Promise<void> {
  return apiFetch<void>(`/config-item/${encodeURIComponent(id)}`, {
    method: 'PUT',
    body: JSON.stringify(req),
  })
}

export function deleteConfigItem(id: string): Promise<void> {
  return apiFetch<void>(`/config-item/${encodeURIComponent(id)}`, {
    method: 'DELETE',
  })
}

File: internal/mcpserver/client/model/configitem.go
package model

type ConfigItem struct {
	ID          string `json:"id"`
	CreatedAt   string `json:"created_at"`
	UpdatedAt   string `json:"updated_at"`
	ConfigmapID string `json:"configmap_id"`
	Active      bool   `json:"active"`
	Key         string `json:"key"`
	Value       string `json:"value"`
	ValueFormat string `json:"value_format"`
	Encoding    string `json:"encoding"`
	FileName    string `json:"file_name"`
	ContentType string `json:"content_type"`
	Description string `json:"description"`
}

// List

type ConfigItemListReq struct {
	ListParams   ListParams
	ConfigmapID  *string
	ConfigmapIDs []string
	Active       *bool
	Search       *string
}

type ConfigItemListRep struct {
	PaginationInfo PaginationInfo `json:"pagination_info"`
	Results        []ConfigItem   `json:"results"`
}

// Create

type ConfigItemCreateReq struct {
	ConfigmapID string `json:"configmap_id"`
	Active      *bool  `json:"active,omitempty"`
	Key         string `json:"key"`
	Value       string `json:"value"`
	ValueFormat string `json:"value_format,omitempty"`
	Encoding    string `json:"encoding,omitempty"`
	FileName    string `json:"file_name,omitempty"`
	ContentType string `json:"content_type,omitempty"`
	Description string `json:"description"`
}

type ConfigItemCreateRep struct {
	ID string `json:"id"`
}

// Update

type ConfigItemUpdateReq struct {
	ConfigmapID *string `json:"configmap_id,omitempty"`
	Active      *bool   `json:"active,omitempty"`
	Key         *string `json:"key,omitempty"`
	Value       *string `json:"value,omitempty"`
	ValueFormat *string `json:"value_format,omitempty"`
	Encoding    *string `json:"encoding,omitempty"`
	FileName    *string `json:"file_name,omitempty"`
	ContentType *string `json:"content_type,omitempty"`
	Description *string `json:"description,omitempty"`
}
